src/main/java/alyssa/Deadline.java

```diff
@@ -0,0 +1,38 @@
+package alyssa;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A Deadline is a task with an end date/time.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```

Comment:
It's not immediately clear that your parameter `by` represents a date. Perhaps a name such as `byDate` or `dueDate` would be more clear to readers? (This also applies to your `LocalDate` parameters in `Event`)